src/main/java/net/mcreator/ui/modgui/OverlayGUI.java

```diff
@@ -95,6 +95,16 @@ public OverlayGUI(MCreator mcreator, ModElement modElement, boolean editingMode)
 		editor.setComponentList(overlay.components);
 		editor.overlayBaseTexture.setSelectedItem(overlay.baseTexture);
 		displayCondition.setSelectedProcedure(overlay.displayCondition);
+
+		editor.sx.setValue(overlay.gridSettings.sx);
+		editor.sy.setValue(overlay.gridSettings.sy);
+		editor.ox.setValue(overlay.gridSettings.ox);
+		editor.oy.setValue(overlay.gridSettings.oy);
+		editor.snapOnGrid.setSelected(overlay.gridSettings.snapOnGrid);

```

Comment:
I think calling this already calls action listeners and runs the if below. Please test if if below is needed and remove accordingly if ok without it.

Reply:
No it does not the same. snapOnGrid isn't the same as showGrid

Reply:
Snap on grid action listener does this:

> snapOnGrid.addActionListener(event -> {
			editor.showGrid = snapOnGrid.isSelected();
			editor.repaint();
		});

This seems same as your if below and it likely called on setSelected..

Reply:
Okay so we really need the `if` part. I tried without it, and when I re-entered the GUI editor, the grid wasn't here. The settings were changed (which is good) but the problem was that the grid was not showing. So we really need the `if`

Reply:
Btw this is actually weird because the checkbox "snap components on the grid" is checked but the grid is not showing so we need to un-check and re-check to get the grid to show.

Reply:
Hmm so actionlistener is not called when not there, than it is ok the way you did it.